app/models/goal.py

```diff
@@ -1,5 +1,34 @@
-from sqlalchemy.orm import Mapped, mapped_column
+from sqlalchemy.orm import Mapped, mapped_column, relationship
 from ..db import db
 
+from typing import TYPE_CHECKING
+if TYPE_CHECKING:
+    from .task import Task
+
+
 class Goal(db.Model):
     id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
+    title: Mapped[str]
+    tasks: Mapped[list["Task"]] = relationship(back_populates="goal")
+
+    def to_dict(self, withRelationship=False):
```

Comment:
```suggestion
    def to_dict(self, with_relationship=False):
```

Prefer snake case since that's what's conventional for Python.